instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -79,46 +80,55 @@ private RequestType getTypeFromAttributes(ExecutionAttributes executionAttribute
   @Override
   public void beforeExecution(
       Context.BeforeExecution context, ExecutionAttributes executionAttributes) {
-    Span span = tracer().getOrCreateSpan(spanName(executionAttributes), AwsSdk.tracer(), kind);
-    executionAttributes.putAttribute(
-        CONTEXT_ATTRIBUTE, io.opentelemetry.context.Context.current().with(span));
+    io.opentelemetry.context.Context parentContext = io.opentelemetry.context.Context.current();
+    if (!tracer().shouldStartSpan(parentContext)) {
+      return;
+    }
+    io.opentelemetry.context.Context otelContext =
+        tracer().startSpan(parentContext, spanName(executionAttributes), AwsSdk.tracer(), kind);
+    executionAttributes.putAttribute(CONTEXT_ATTRIBUTE, otelContext);
     RequestType type = ofSdkRequest(context.request());
     if (type != null) {
       executionAttributes.putAttribute(REQUEST_TYPE_ATTRIBUTE, type);
     }
+    if (executionAttributes
+        .getAttribute(SdkExecutionAttribute.CLIENT_TYPE)
+        .equals(ClientType.SYNC)) {
+      // We can only activate context for synchronous clients, which allows downstream
+      // instrumentation like Apache to know about the SDK span.
+      executionAttributes.putAttribute(SCOPE_ATTRIBUTE, otelContext.makeCurrent());
+    }
   }
 
   @Override
   public SdkHttpRequest modifyHttpRequest(
       Context.ModifyHttpRequest context, ExecutionAttributes executionAttributes) {
-    io.opentelemetry.context.Context otelContext =
-        executionAttributes.getAttribute(CONTEXT_ATTRIBUTE);
-    // Never null in practice unless another interceptor cleared out the attribute, which
-    // is theoretically possible.
+    io.opentelemetry.context.Context otelContext = getContextFromAttributes(executionAttributes);
     if (otelContext == null) {
       return context.httpRequest();
     }
+
     SdkHttpRequest.Builder builder = context.httpRequest().toBuilder();
-    OpenTelemetry.getGlobalPropagators()
-        .getTextMapPropagator()
-        .inject(otelContext, builder, AwsSdkInjectAdapter.INSTANCE);
+    AwsXRayPropagator.getInstance().inject(otelContext, builder, AwsSdkInjectAdapter.INSTANCE);
```

Comment:
Even if you are sure that AWS SDK calls are always terminal and that nobody uses some alternative implementations of AWS services (like local S3 server?), we at least have to document this behaviour very clearly.

Reply:
Got a bit of extra README out of it too :)